scripts/assets/options.go.templ

```diff
@@ -109,6 +109,8 @@ func UpdateFieldsForJSON(flags *pflag.FlagSet) (*serverSettingsForJSON, error) {
 		}
 	<% } %>
 	if changedSomething {
+		specifiedSettings.Version = <%- settingsVersion %>
+		specifiedSettingsForJSON.Version = &specifiedSettings.Version;
```

Comment:
To answer my question from https://github.com/rancher-sandbox/rancher-desktop/pull/3890#discussion_r1101851177: `specifiedSettingsForJSON.Version` is a pointer, and we needed update the underlying value too (hence changing `specifiedSettings.Version`).

Reply:
I covered this back in the PR to generate options in the first place.

I could have declared various registers to capture changed values and then set pointers to them in the JSON payload, but it's more straightforward to just declare a shadow structure with typed fields rather, and just point to them.